Rakefile

```diff
@@ -20,14 +20,23 @@ Bundler::GemHelper.install_tasks
 
 require "rake/testtask"
 
-Rake::TestTask.new(:test) do |t|
-  t.libs = ["lib", "test"]
-  t.ruby_opts = ["-rsetup"]
-  t.pattern = "test/**/*_test.rb"
-  t.verbose = false
-
-  # Hide warnings emitted by our dependencies.
-  t.warning = false
+namespace :test do
+  ["postgresql", "mysql2"].each do |adapter|
+    adapter_short = adapter[/^[a-z]+/]
+    Rake::TestTask.new(adapter_short => "set_#{adapter_short}_env") do |t|
+      t.libs = ["lib", "test"]
```

Comment:
I'd stick to the public API of `Rake::TestTask` which indicates the constructor expects `name` as the only parameter. We can pass deps via `t.deps = ...`.

Reply:
Than seems like a public API, I think - https://github.com/ruby/rake/blob/fc688bd8cd6b4178178a0d7bb31068122ba25943/lib/rake/testtask.rb#L98-L101

But nevertheless, changed to `t.deps = ...`